lib/Bio/EnsEMBL/DataCheck/Checks/MTCodonTable.pm

```diff
@@ -0,0 +1,66 @@
+=head1 LICENSE
+
+Copyright [2018-2019] EMBL-European Bioinformatics Institute
+
+Licensed under the Apache License, Version 2.0 (the 'License');
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an 'AS IS' BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+=cut
+
+package Bio::EnsEMBL::DataCheck::Checks::MTCodonTable;
+
+use warnings;
+use strict;
+
+use Moose;
+use Test::More;
+use Bio::EnsEMBL::DataCheck::Utils qw/sql_count/;
+use Bio::EnsEMBL::DataCheck::Test::DataCheck;
+
+extends 'Bio::EnsEMBL::DataCheck::DbCheck';
+
+
+use constant {
+  NAME           => 'MTCodonTable',
+  DESCRIPTION    => 'MT seq region had associated seq_region attribute '
+    . 'and correct codon table',
+  GROUPS         => ['core_handover'],
+  DATACHECK_TYPE => 'critical',
+  DB_TYPES       => ['cdna', 'core', 'funcgen', 'otherfeatures', 'rnaseq', 'variation'],
+  TABLES         => ['seq_region', 'seq_region_attrib']
+};
+
+sub tests {
+  my ($self) = @_;
+
+  my $desc_1 = 'MT seq_regions have associated seq_region_attrib '
+    . 'using correct codon table';
+  my $sql_1 = q/
+      SELECT COUNT(seq_region_id)
+        FROM seq_region
+       WHERE name like '%MT%'
```

Comment:
This query is not "collection-safe" (cf https://www.ebi.ac.uk/seqdb/confluence/display/GTI/Everything+you+wanted+to+know+about+datachecks%2C+but+were+too+afraid+to+ask#Everythingyouwantedtoknowaboutdatachecks,butweretooafraidtoask-Collectiondatabases).
It needs to link to coord_system and filter on species_id.
Also '%MT%' is perhaps too permissive; it's not too much of a stretch to imagine a non-MT seq that just has 'MT' in the name some where. Maybe '%MT'?

Reply:
I'm not sure comparing counts is the right thing to do here. I know that's what the HC did, but why not replace the two queries with a single one that returns a list of seq_region_ids and names where the name matches MT, but the attrib is missing, _a la_ https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/SeqRegionTopLevel.pm#L59

Reply:
Or, one step further, you could use the API, something like https://github.com/Ensembl/ensembl-datacheck/blob/master/lib/Bio/EnsEMBL/DataCheck/Checks/ChromosomesAnnotated.pm#L56